Docs - Components props tables are missing

### Prerequisites

- [X] I am using the [correct version](https://github.com/react-bootstrap/react-bootstrap#bootstrap-compatibility) of React-Bootstrap for my version of Bootstrap
- [X] I have [searched](https://github.com/react-bootstrap/react-bootstrap/issues?q=is%3Aissue) for duplicate or closed issues
- [X] I have read the [contributing guidelines](https://github.com/react-bootstrap/react-bootstrap/blob/master/CONTRIBUTING.md)

### Describe the bug

On the documentation ([v2.10.0](https://react-bootstrap.netlify.app/docs/components)), the props tables are missing from the API section.

### Expected behavior

The API section should display the components props tables listing the props with their type, default value and description.

### To Reproduce

1. Go to a component's documentation page (ex: https://react-bootstrap.netlify.app/docs/components/accordion).
2. Scroll down to the API section.
3. Only the components names are shown.

### Reproducible Example

https://react-bootstrap.netlify.app/docs/components/accordion#api

### Screenshots

![Capture d’écran du 2024-02-05 13-32-13](https://github.com/react-bootstrap/react-bootstrap/assets/36835658/5ebb5653-b986-411f-9afd-7f42693e2ca8)
![Capture d’écran du 2024-02-05 13-32-52](https://github.com/react-bootstrap/react-bootstrap/assets/36835658/a6f20146-5532-4dd4-ba33-b36c642963a8)


### What operating system(s) are you seeing the problem on?

Linux

### What browser(s) are you seeing the problem on?

Chrome, Firefox

### What version of React-Bootstrap are you using?

2.10.0

### What version of Bootstrap are you using?

NA

### Additional context

Errors are thrown in the console stating `Cannot find module './<component's name>.json'` (see screenshot above).

--- a/www/package.json
+++ b/www/package.json
@@ -33,7 +33,7 @@
     "prism-react-renderer": "^2.1.0",
     "raw-loader": "^4.0.2",
     "react": "^18.2.0",
-    "react-docgen": "^7.0.3",
+    "react-docgen": "^5.4.3",
     "react-dom": "^18.2.0",
     "react-live": "^2.4.1",
     "sass": "^1.69.7",

--- a/www/plugins/react-docgen-plugin/index.js
+++ b/www/plugins/react-docgen-plugin/index.js
@@ -29,8 +29,7 @@ module.exports = () => ({
         const filePath = path.join(srcPath, file);
         const buffer = await fs.readFile(filePath);
 
-        const output = reactDocgen.parse(buffer, {
-          resolver: resolveHocComponents,
+        const output = reactDocgen.parse(buffer, resolveHocComponents, null, {
           filename: filePath,
         });
 

--- a/www/plugins/react-docgen-plugin/resolveHocComponents.js
+++ b/www/plugins/react-docgen-plugin/resolveHocComponents.js
@@ -1,9 +1,14 @@
 const camelCase = require('lodash.camelcase');
 const { resolver, utils } = require('react-docgen');
 const { namedTypes: types, visit } = require('ast-types');
-const { parse } = require('react-docgen');
+const buildParser = require('react-docgen/dist/babelParser').default;
 
 const upperFirst = (str) => str[0].toUpperCase() + str.slice(1);
+
+const parser = buildParser({
+  // Avoid babel complaining about requiring a filename.
+  filename: 'dummy.ts',
+});
 
 module.exports = (ast) => {
   let components = resolver.findAllComponentDefinitions(ast);
@@ -71,7 +76,7 @@ export default class ${upperFirst(
 }
         `;
 
-      let comp = parse(src);
+      let comp = parser.parse(src);
       comp.__src = src;
       components = components.concat(
         resolver.findExportedComponentDefinition(comp),

